src/HtmlReplacer.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Astrotomic\Twemoji;
+
+use Astrotomic\Twemoji\Concerns\Configurable;
+use RuntimeException;
+use Wa72\HtmlPageDom\HtmlPage;
+use Wa72\HtmlPageDom\HtmlPageCrawler;
+
+class HtmlReplacer
+{
+    use Configurable;
+
+    public static string $shouldNotBeParsed = "/^(?:iframe|noframes|noscript|script|select|style|textarea)$/";
+
+    public function __construct()
+    {
+        if (! class_exists(HtmlPageCrawler::class)) {
+            throw new RuntimeException(
+                sprintf('Cannot use %s method unless `wa72/htmlpagedom` is installed.', __METHOD__)
+            );
+        }
+    }
+
+    public function parse(string $html): string
```

Comment:
If we need to support full HTML docs and HTML fragments, then this method should:
1. Immediately determine if the input `$html` is a full DOM page, then
2. either use HtmlPage (used here) and work based on the Body, or
3. use the HtmlPageCrawler to parse the fragment.

Reply:
I think that in PHP partial HTML is more common than a full document. Except you are implementing it as some kind of middleware to parse the whole HTML response.
But in general it should support both if possible.

Reply:
Addressed this by using the more general HTML parser, then adding a step where we check if the input HTML is a Page/Doc and selecting the `body` from that. As I was already replacing based on a HTML fragment (the body), supporting fragments as input was rather simple.